Observer/ReleaseProductAdd.php

```diff
@@ -0,0 +1,61 @@
+<?php
+declare(strict_types=1);
+
+namespace PayPal\Subscription\Observer;
+
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Exception\NotFoundException;
+use Magento\Quote\Api\Data\CartInterface;
+use Magento\Quote\Model\Quote\Item;
+use Magento\Tax\Api\TaxCalculationInterface;
+use Magento\Tax\Helper\Data as TaxHelper;
+
+class ReleaseProductAdd implements ObserverInterface
+{
+    /**
+     * @param TaxHelper $taxHelper
+     * @param TaxCalculationInterface $taxCalculation
+     */
+    public function __construct(
+        private readonly TaxHelper $taxHelper,
+        private readonly TaxCalculationInterface $taxCalculation
+    ) {}
+
+    /**
+     * @inheritDoc
+     */
+    public function execute(Observer $observer)
+    {
+        /** @var Item $item */
+        $item = $observer->getEvent()->getData('quote_item');
+
+        if (!$item->getQuote() instanceof CartInterface ||
+            (bool) $item->getQuote()->getData('is_subscription_release') !== true
+        ) {
+            return;
+        }
+
+        $product = $item->getProduct();
+        if (!$product instanceof ProductInterface) {
+            throw new NotFoundException(__("Could not find product for quote item ID %1", $item->getItemId()));
+        }
+
+        // Product price is taken from subscription price, meaning it is already inclusive of discount.
+        $subscriptionPrice = (float) $product->getPrice();
```

Comment:
@AlexRapatij So this code is partially taken from the FE add to cart observer. In that observer, the subscription price is added to the quote item as a custom price (same is done here): https://github.com/genecommerce/paypal-subscription-module/blob/master/Observer/CartProductAddAfterObserver.php#L109

The reason this uses `getPrice` function is because the subscription price is set to the product `price` field when creating the quote from subscription items here: https://github.com/genecommerce/paypal-subscription-module/blob/master/Model/CreateSubscriptionQuote.php#L267-L269. This product is then passed to core `addProduct` which in turn is passed to this event observer.